src/deepsparse/loggers/function_logger.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation of the Function Logger
+"""
+
+from typing import Any, Dict, List
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+from deepsparse.loggers.metric_functions import apply_function
+
+
+__all__ = ["FunctionLogger"]
+
+
+class FunctionLogger(BaseLogger):
+    """
+    DeepSparse logger that applies functions to raw log values
+    according to the specified config file
+
+    :param loggers: A DeepSparse Logger object
+    :param config: A configuration dictionary that specifies the mapping between
+        the target name and the functions to be applied to raw log values
+        e.g.
+        ```
+        {"pipeline_inputs":
+            [{"function": "identity",
+              "frequency": 3},
+
+            [{"function": "identity",
+              "frequency": 5}],
+
+         "pipeline_outputs":
+            [{"function": "identity",
+            "frequency": 4}]
+        }
+    """
+
+    def __init__(
+        self,
+        logger: BaseLogger,
+        config: Dict[str, List[Dict[str, Any]]],
+    ):
+
+        self.logger = logger
+        self.config = self._add_frequency_counter(config)
```

Comment:
let's avoid in-place modifying the user passed config, either create a copy or we should track frequency elsewhere (ie we only need to measure number of times an identifier has been triggered in one location rather than for each of its functions)